### ISSUE TITLE
Validation failing when inserting a new phase

### ISSUE THREAD

**Issue by DBazin** [ATTPC/Spyral#128]
The schema validation fails when inserting a new phase inside the pipeline, because the name of the schema expected by the phase following the new one doesn't match the schema's name of the inserted new phase, even if they have the same structure. 
Maybe this is moot because in most cases the phases following the inserted one will have to be customized anyway if the inserted phase changes the structure of the schema being transmitted. Something to discuss I think.

**Comment by gwm17** [comment]
Yeah this is definitely worth looking at. The schema is pretty brittle the way I put it together now, particularly for inserting a Phase. For the issue you've described, was it that the structure (the dictionary/list in the schema) or the extension that was giving you problems?

**Comment by DBazin** [comment]
The schema of my inserted phase is actually the same structure as the previous one (ESTIMATE_SCHEMA), I just changed the name, so the validation seems to rely on that name rather than the structure itself.

**Comment by gwm17** [comment]
That's weird... I overloaded the equality comparison operator to specifically only compare the individual fields. I'll look into this more carefully and see what's going on!

**Comment by gwm17** [comment]
Ah ok I think I see the problem. I'm comparing two lists (or dictionaries) which is not a well defined op with == by default. Should be relatively easy to fix. Will be slated for the next release!

**Comment by gwm17** [comment]
Removing the bug label, but leaving this open as it is still being discussed

**Comment by gwm17** [comment]
Marking this as closed for now. I think the new solution is better, but still not perfect.
